addons/stock/models/stock_scrap.py

```diff
@@ -47,11 +37,13 @@ def _get_default_location_id(self):
     move_id = fields.Many2one('stock.move', 'Scrap Move', readonly=True, check_company=True, copy=False)
     picking_id = fields.Many2one('stock.picking', 'Picking', states={'done': [('readonly', True)]}, check_company=True)
     location_id = fields.Many2one(
-        'stock.location', 'Source Location', domain="[('usage', '=', 'internal'), ('company_id', 'in', [company_id, False])]",
-        required=True, states={'done': [('readonly', True)]}, default=_get_default_location_id, check_company=True)
+        'stock.location', 'Source Location',
+        compute='_compute_location_id', store=True, required=True, precompute=True, states={'done': [('readonly', True)]},
+        domain="[('usage', '=', 'internal'), ('company_id', 'in', [company_id, False])]", check_company=True)
     scrap_location_id = fields.Many2one(
-        'stock.location', 'Scrap Location', default=_get_default_scrap_location_id,
-        domain="[('scrap_location', '=', True), ('company_id', 'in', [company_id, False])]", required=True, states={'done': [('readonly', True)]}, check_company=True)
+        'stock.location', 'Scrap Location',
+        compute='_compute_scrap_location_id', store=True, required=True, precompute=True, states={'done': [('readonly', True)]},
+        domain="[('scrap_location', '=', True), ('company_id', 'in', [company_id, False])]", check_company=True)
```

Comment:
Since all of the location changes are unrelated to the uom fix, it would be cleaner to put them in a separate commit with the `[REF]` tag so the commit history is cleaner/easier to revert if necessary in the future.

Reply:
done